File: fontes/senha.cpp
// Teste para uso de senha
// Programa didático, não pode ser
// usado profissionalmente

#include <iostream>

int T = 3;
bool CORRETO = false;
int SENHA;

int main(void) 
{
  
  do 
   {
      std::cout << "Entre a senha de acesso: ";
      std::cin >> SENHA;
      if (SENHA == 123)
        CORRETO = true;
      else 
       {
         T = T - 1;
         std::cout << "Tentativas restantes: " << T << std::endl;
       }
    } 
  while (not(CORRETO == true or T == 0));    
  
  if (CORRETO == true)
    std::cout << "Acesso liberado" << std::endl;
  else
    std::cout << "Acesso bloqueado" << std::endl;
  
  return 0;
}

File: fontes/senha.vb
' Teste para uso de senha
' Programa didático, não pode ser
' usado profissionalmente
' 
' Linguagem BASIC (VB.Net)

Module MainModule
  Sub Main()

    Dim T As Integer = 3
    Dim CORRETO As Boolean = False
    Dim SENHA As Integer = 0

    Do
      Console.Write("Entre a senha de acesso: ")
      SENHA = Integer.Parse(Console.ReadLine())
      If SENHA = 123 Then
        CORRETO = True
      Else
        T = T - 1
        Console.WriteLine("Tentativas restantes: " & T)
      End If
    Loop Until (CORRETO = True Or T = 0)

    If (CORRETO = True) Then
      Console.WriteLine("Acesso liberado")
    Else
      Console.WriteLine("Acesso bloqueado")
    End If
    Console.ReadLine()

  End Sub
End Module
